athenzdomains.go

```diff
@@ -0,0 +1,183 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/yahoo/athenz/clients/go/zms"
+	v1 "github.com/yahoo/k8s-athenz-istio-auth/pkg/apis/athenz/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	replacer = strings.NewReplacer(".*", ".*", "*", ".*")
+)
+
+// CRMap - cr cache
+type CRMap struct {
+	RoleToPrincipals map[string][]*zms.RoleMember
```

Comment:
We could store the member regex here directly to avoid the compilation during the authorize function call.